### PR TITLE
added text change width and shape of button

### PR DESCRIPTION
**Pull request by markarik**
From this link on [Google Button Generator](https://developers.google.com/identity/gsi/web/tools/configurator)
there are attributes from the generated code which would be essential for this package to make it atlist customizable as much as possible

Below is a screenshot of the demo page after working on the attributes

Screenshot (38)

### CODE REVIEW

**Comment by kattoshi** [comment]
I would like to request the following items
`theme`
`logo_alignmnt`
`locale`

**Comment by markarik** [comment]
For those am still testing on them, once comfortable will push ASAP

**Comment by markarik** [comment]
@kattoshi  Have managed to add `theme `and  `logo_alignment`  
unfortunately on `locale` have tried testing it but not working on it,
Have opted to leave it as   from the [docs](https://developers.google.com/identity/gsi/web/reference/html-reference#data-locale) it's written if the locale is left as empty, the button should pick the browsers default locale

**Comment by Heatmanofurioso** [comment]
@markarik Thanks a lot for your contribution here. Would this be fine to be merged by you? Or would you try to add any more properties to it? The PR looks good by me

File: projects/lib/src/directives/google-signin-button.directive.ts [hunk 1]
```diff
@@ -8,20 +8,59 @@ import { SocialAuthService } from '../socialauth.service';
...
-      google.accounts.id.renderButton(el.nativeElement, {
-        type: this.type,
-        size: this.size,
+
+
+      Promise.resolve(this.locale).then((value) => {
+
+        console.log("value", value);
+      })
+
```
Comment (kattoshi) [inline projects/lib/src/directives/google-signin-button.directive.ts:42]: degrades performance

**Comment by kattoshi** [comment]
@markarik Thank you. the test is working fine.
image
image
image

File: src/app/demo/demo.component.html [hunk 2]
```diff
@@ -5,8 +5,13 @@ <h6 class="card-header">Social Login Demo</h6>
       <h4 class="card-title">Not signed in</h4>
       <p class="card-text">Sign in with</p>
     </div>
+
+    
+
     <div class="card-block d-flex justify-content-center">
-      <asl-google-signin-button class="mx-1"></asl-google-signin-button>
+
+      <asl-google-signin-button class="mx-5"></asl-google-signin-button>
```
Comment (kattoshi) [inline src/app/demo/demo.component.html:13]: I think "mx-1" is better

**Review by kattoshi — COMMENTED** [review]
Isn't 'package-lock.json' unnecessary in PR?

**Comment by markarik** [comment]
> @markarik Thanks a lot for your contribution here. Would this be fine to be merged by you? Or would you try to add any more properties to it? The PR looks good by me

Am humbled,, I think am not in a position to merge it

**Comment by markarik** [comment]
> @markarik Thank you. the test is working fine. image image image

Great thanks ,, I suppose on my laptop it just dint want to work at the moment

**Comment by Heatmanofurioso** [comment]
> > @markarik Thanks a lot for your contribution here. Would this be fine to be merged by you? Or would you try to add any more properties to it? The PR looks good by me
> 
> Am humbled,, I think am not in a position to merge it

I can merge it. What I meant was to ask if you intended to make any more commits to this PR.
Since you've made some changes since it's initial opening

**Comment by markarik** [comment]
> > > @markarik Thanks a lot for your contribution here. Would this be fine to be merged by you? Or would you try to add any more properties to it? The PR looks good by me
> > 
> > 
> > Am humbled,, I think am not in a position to merge it
> 
> I can merge it. What I meant was to ask if you intended to make any more commits to this PR. Since you've made some changes since it's initial opening

Yes you can merge it, I don't intend to add anything else

**Pull request merged**